lib/kafka_ex.ex

```diff
@@ -144,14 +144,10 @@ defmodule KafkaEx do
     max_bytes     = Keyword.get(opts, :max_bytes, @max_bytes)
     auto_commit   = Keyword.get(opts, :auto_commit, true)
 
-    offset = case offset do
-      nil -> last_offset = offset_fetch(worker_name, %KafkaEx.Protocol.OffsetFetch.Request{topic: topic, partition: partition}) |> KafkaEx.Protocol.OffsetFetch.Response.last_offset
-        if last_offset <= 0 do
-          0
-        else
-          last_offset
-        end
-      _   -> offset
+    unless offset do
+      # retrieve last stored offset from kafka
+      last_offset = offset_fetch(worker_name, %KafkaEx.Protocol.OffsetFetch.Request{topic: topic, partition: partition}) |> KafkaEx.Protocol.OffsetFetch.Response.last_offset
+      offset = max(last_offset + 1, 0)
```

Comment:
if `last_offset` returns 0, then that means that another process consumed message 0 and committed that offset, so 1 in that case would be right? i'm new to kafka, so apologies if i'm missing something

>  I prefer the case over if/unless

For this specific piece of code, I actually find it difficult to read because it requires a scan of 7 additional lines of code to see that `offset`'s value isn't changed if it's not nil; the `unless` makes this immediately apparent. I don't want to make this a point of contention though and will use `case` if that's what you want